jax/_src/random.py

```diff
@@ -628,23 +628,21 @@ def normal(key: jnp.ndarray,
 @partial(jit, static_argnums=(1, 2))
 def _normal(key, shape, dtype) -> jnp.ndarray:
   if dtypes.issubdtype(dtype, np.complexfloating):
-    sqrt2 = np.array(np.sqrt(2), dtype)
-
     key_re, key_im = split(key)
     dtype = dtypes.dtype_real(dtype)
     _re = _normal_real(key_re, shape, dtype)
     _im = _normal_real(key_im, shape, dtype)
-    return 1 / sqrt2 * (_re + 1j * _im)
```

Comment:
the 1 here was the culprit.

Reply:
I think it might be confusing to move the sqrt2 factor out of the helper function, because it no longer computes what it says it computes (`normal_real`). Instead, you could get around the numpy type promotion issue by making this line
```python
(_re + 1j * _im) / sqrt2
```
and keeping everything else the same. Since this is JIT-compiled, the multiplication and division by the same constant will be optimized out of the final computation.

Reply:
done. I am amazed that these kinds of constants can be completely optimized away :D